### DISCUSSION TITLE
Manage permissions to add or delete columns

### DISCUSSION THREAD

**Discussion by slasia** [AntlerVC/firetable#1161] [Support / Q&A] [answered]
Hello! I wanted to ask if there is some mechanism in the CMS for a user to have permissions to:

- Add rows
- Delete rows
- Edit row content

and do not have permissions to:

- Add new column
- Delete current column

Thanks!

**Comment by BeeBombshell** [comment, marked as answer]
Hey @slasia, you'll need to go to the Workspace Settings, click on Invite User, and under the Roles dropdown, click on Add New. Add the role name, let's say, **MEMBER**. Invite the users using their email ids.

Now, you need to navigate to the Firestore Rules page (here: https://console.firebase.google.com/project/%7Bproject-id%7D/firestore/rules), and append the following rules:
```
rules_version = '2';
service cloud.firestore {
  match /databases/{database}/documents {
    match /collection/{document=**} {
      allow read: if true;
      allow create: if request.auth != null && request.auth.token.roles['MEMBER'] == true;
      allow update, delete: if request.auth != null && request.auth.token.roles['MEMBER'] == true &&
        (
          request.resource.data.keys().hasAll(resource.data.keys()) &&
          request.resource.data.size() == resource.data.size()
        );
    }
  }
}
```
Hope this helps!
